index.bs

```diff
@@ -205,9 +207,22 @@ ISSUE: Clarify the behavior of inline sessions
 
 Automatic composition {#automatic-composition}
 ---------
-<dfn>performs automatic composition</dfn>
 
-ISSUE: Add explanation here
+An [=xr device=] during an {{XRSessionMode/"immersive-ar"}} session will have its [=xr compositor=] <dfn>perform automatic composition</dfn> by drawing the rendered scene on top of the real-world background.
+
+In case the [=xr device=] is using a [=passthrough camera=], it MUST alpha-blend non-opaque rendered pixels on top of the [=passthrough camera=]'s [=aligned video stream=]. Such a device MUST have an {{XRSession/environmentBlendMode}} of {{alpha-blend}}.
+
+If the [=xr device=] is an additive light device where the real-world background is visible through the device's transparent display, it MUST ignore alpha values when performing automatic composition, instead, it MUST treat black pixels as transparent.
```

Comment:
One nuance here is when the app's rendered views are used not just for the primary additive-light displays, but also for a secondary system-managed video capture stream that uses alpha blending.

For example, HoloLens has an additive-light display and does indeed ignore alpha values when performing automatic composition **for the primary display**.  However, [a HoloLens app's alpha channel **is** used](https://docs.microsoft.com/en-us/windows/mixed-reality/mixed-reality-capture-for-developers#best-practices-hololens-specific) when the user enables Mixed Reality Capture at a system level to take a photo or video of what they're seeing with their virtual content composited in.  When the system composits that secondary camera stream, the app's alpha channel controls each pixel's blend between the virtual content and the camera feed of the real world.

Perhaps a more nuanced statement here would be:
> If the [=xr device=] is an additive light device where the real-world background is visible through the device's transparent display, it MUST ignore alpha values **and treat black pixels as transparent when performing automatic composition for the primary displays. Secondary system usage of the app's views, such as a photo capture or video capture stream, MAY make use of the alpha channel.**

We should also nuance the `"additive"` bullet above on line 198:
>   - A blend mode of <dfn enum-value for="XREnvironmentBlendMode">additive</dfn> indicates that the user's surrounding environment is visible and the {{XRRenderState/baseLayer}} will be shown additively against it **on the device's primary displays**. **On those displays**, alpha values in the {{XRRenderState/baseLayer}} will be ignored, with the compositor treating all alpha values as 1.0. **This means that** black pixels will appear fully transparent, and there is no way to make a pixel appear fully opaque. **Secondary system usage of the app's views, such as a photo capture or video capture stream, MAY make use of the alpha channel.**

Reply:
addressed